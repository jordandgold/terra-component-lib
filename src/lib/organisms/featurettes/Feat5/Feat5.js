import React, { Component } from "react";
import Dumpling from "../../../Dumpling/Dumpling";
import PropTypes from "prop-types";
import "./Feat5.scss";

class Feat5 extends Component {
  getDumplings = () => {
    return this.props.dumplings.map((dumpling, index) => {
      if (dumpling.icon) {
        return (
          <div
            className={`ter-feat-five__dumpling-wrapper ter-feat-five__dumpling-wrapper--${
              this.props.type
            }`}
            key={`dumpling-${index}`}
          >
            <Dumpling
              title={dumpling.title}
              size={this.props.type === "b" ? "small" : "large"}
              link={dumpling.link}
              icon={dumpling.icon}
            />
          </div>
        );
      } else if (dumpling.image) {
        return (
          <div
            className={`ter-feat-five__dumpling-wrapper ter-feat-five__dumpling-wrapper--${
              this.props.type
            }`}
            key={`dumpling-${index}`}
          >
            <Dumpling
              title={dumpling.title}
              size={this.props.type === "b" ? "small" : "large"}
              link={dumpling.link}
              image={dumpling.image}
            />
          </div>
        );
      }
    });
  };

  render() {
    const { title, subtitle } = this.props;
    return (
      <section className="ter-feat-five">
        <h2 className="ter-feat-five__title">{title}</h2>
        {subtitle && <h4 className="ter-feat-five__subtitle">{subtitle}</h4>}
        <div className="ter-feat-five__dumpling-container">
          {this.getDumplings()}
        </div>
      </section>
    );
  }
}

export default Feat5;

Feat5.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  type: PropTypes.string.isRequired,
  dumplings: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string.isRequired,
      link: PropTypes.shape({ url: PropTypes.string.isRequired }).isRequired
    })
  )
};
