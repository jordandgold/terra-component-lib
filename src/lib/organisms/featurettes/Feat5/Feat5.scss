@import "../../../global-styles/terra.scss";

.ter-feat-five {
  padding-top: spacing("md");

  &__title {
    font-size: font-size("xl");
  }

  &__subtitle {
    font-weight: font-weight("light");
    color: color("gray-7");
  }

  &__subtitle,
  &__title {
    text-align: center;
  }

  &__dumpling-container {
    @include grid-container;
    margin-top: spacing("md");
  }

  &__dumpling-wrapper--b {
    @include grid-column(3);

    @media (max-width: 900px) {
      @include grid-column(6);
    }

    @media (max-width: 600px) {
      @include grid-column(12);
    }
  }

  &__dumpling-wrapper--a {
    @include grid-column(4);

    @media (max-width: 900px) {
      @include grid-column(12);
    }

    @media (max-width: 600px) {
      @include grid-column(12);
    }
  }
}

@media (max-width: 1200px) {
  .ter-feat-five {
    &__title {
      font-size: font-size("lg");
    }

    .ter-icon {
      // transform: scale(0.66666);

      // LEFT OUT PER REC FROM JORDAN
    }
  }
}

@media (max-width: 600px) {
  .ter-feat-five {
    &__dumpling-container {
      justify-content: space-around;
    }
  }
}
