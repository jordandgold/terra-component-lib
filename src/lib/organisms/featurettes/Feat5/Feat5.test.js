import React from "react";
import { shallow } from "enzyme";
import Feat5 from "./Feat5";
import {
  mockTestIconDumplings,
  mockTestImageDumplings
} from "../../../../stories/mockContent";

describe("Feat5", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Feat5
        dumplings={mockTestIconDumplings}
        type="b"
        title="Design is design"
        subtitle="It's like, this thing, you know?"
      />
    );
  });

  it("should match the snapshot with icons", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should match the snapshot with images", () => {
    wrapper = shallow(
      <Feat5
        dumplings={mockTestImageDumplings}
        type="b"
        title="Design is design"
        subtitle="It's like, this thing, you know?"
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  describe("getDumplings", () => {
    it("should return some jsx if icons", () => {
      const expected = [
        '<div class="ter-feat-five__dumpling-wrapper ter-feat-five__dumpling-wrapper--b"><article class="ter-dumpling ter-dumpling--small"><div class="ter-dumpling__icon-wrapper"><svg viewBox="0 0 8 8" class="ter-icon ter-icon--enclosed-check-dark-48px ter-icon--48px undefined"><use xlink:href="[object Object]#ter-icon--enclosed-check-dark-48px"></use></svg></div><div class="ter-dumpling__small-dash"></div><a class="ter-dumpling__link ter-dumpling__link--small" href="#">Dumplin&#x27;</a><div></div></article></div>',
        '<div class="ter-feat-five__dumpling-wrapper ter-feat-five__dumpling-wrapper--b"><article class="ter-dumpling ter-dumpling--small"><div class="ter-dumpling__icon-wrapper"><svg viewBox="0 0 8 8" class="ter-icon ter-icon--enclosed-check-dark-48px ter-icon--48px undefined"><use xlink:href="[object Object]#ter-icon--enclosed-check-dark-48px"></use></svg></div><div class="ter-dumpling__small-dash"></div><a class="ter-dumpling__link ter-dumpling__link--small" href="#">Dumplin&#x27;</a><div></div></article></div>'
      ];
      const response = wrapper
        .instance()
        .getDumplings()
        .map(dumpling => {
          return shallow(dumpling).html();
        });

      expect(response).toEqual(expected);
    });

    it("should return some jsx if images", () => {
      const expected = [
        '<div class="ter-feat-five__dumpling-wrapper ter-feat-five__dumpling-wrapper--b"><article class="ter-dumpling ter-dumpling--small"><img src="./1-to-1.png" alt="test text" class="ter-dumpling__image"/><div class="ter-dumpling__small-dash"></div><a class="ter-dumpling__link ter-dumpling__link--small" href="#">Dumplin&#x27;</a><div></div></article></div>',
        '<div class="ter-feat-five__dumpling-wrapper ter-feat-five__dumpling-wrapper--b"><article class="ter-dumpling ter-dumpling--small"><img src="./1-to-1.png" alt="test text" class="ter-dumpling__image"/><div class="ter-dumpling__small-dash"></div><a class="ter-dumpling__link ter-dumpling__link--small" href="#">Dumplin&#x27;</a><div></div></article></div>'
      ];

      wrapper = shallow(
        <Feat5
          dumplings={mockTestImageDumplings}
          type="b"
          title="Design is design"
          subtitle="It's like, this thing, you know?"
        />
      );

      const response = wrapper
        .instance()
        .getDumplings()
        .map(dumpling => {
          return shallow(dumpling).html();
        });

      expect(response).toEqual(expected);
    });
  });
});
