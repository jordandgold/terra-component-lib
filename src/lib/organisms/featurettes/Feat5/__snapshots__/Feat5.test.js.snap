// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feat5 should match the snapshot with icons 1`] = `
<section
  className="ter-feat-five"
>
  <h2
    className="ter-feat-five__title"
  >
    Design is design
  </h2>
  <h4
    className="ter-feat-five__subtitle"
  >
    It's like, this thing, you know?
  </h4>
  <div
    className="ter-feat-five__dumpling-container"
  >
    <div
      className="ter-feat-five__dumpling-wrapper"
      key="dumpling-0"
    >
      <Dumpling
        icon={
          Object {
            "size": "48px",
            "type": "enclosed-check-dark-48px",
          }
        }
        link={
          Object {
            "url": "#",
          }
        }
        size="small"
        title="Dumplin'"
      />
    </div>
    <div
      className="ter-feat-five__dumpling-wrapper"
      key="dumpling-1"
    >
      <Dumpling
        icon={
          Object {
            "size": "48px",
            "type": "enclosed-check-dark-48px",
          }
        }
        link={
          Object {
            "url": "#",
          }
        }
        size="small"
        title="Dumplin'"
      />
    </div>
  </div>
</section>
`;

exports[`Feat5 should match the snapshot with images 1`] = `
<section
  className="ter-feat-five"
>
  <h2
    className="ter-feat-five__title"
  >
    Design is design
  </h2>
  <h4
    className="ter-feat-five__subtitle"
  >
    It's like, this thing, you know?
  </h4>
  <div
    className="ter-feat-five__dumpling-container"
  >
    <div
      className="ter-feat-five__dumpling-wrapper"
      key="dumpling-0"
    >
      <Dumpling
        image={
          Object {
            "altText": "test text",
            "url": "./1-to-1.png",
          }
        }
        link={
          Object {
            "url": "#",
          }
        }
        size="small"
        title="Dumplin'"
      />
    </div>
    <div
      className="ter-feat-five__dumpling-wrapper"
      key="dumpling-1"
    >
      <Dumpling
        image={
          Object {
            "altText": "test text",
            "url": "./1-to-1.png",
          }
        }
        link={
          Object {
            "url": "#",
          }
        }
        size="small"
        title="Dumplin'"
      />
    </div>
  </div>
</section>
`;
